index.js

```diff
@@ -16,46 +16,112 @@ class ExecutorRouter extends Executor {
      */
     constructor(config = {}) {
         const ecosystem = config.ecosystem || {};
-        const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const { executor, defaultPlugin } = config;
 
-        if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
+        if (!executor || !Array.isArray(executor) || executor.length === 0) {
             throw new Error('No executor config passed in.');
         }
-
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        let ExecutorPlugin;
+
+        this._executors = [];
```

Comment:
this._executors = executor ?

Reply:
configuration could be incorrect and this._executors will have incorrect plugin added, hence adding only after plugin is loaded.